src/parser/RdfParser.cpp

```diff
@@ -529,49 +528,54 @@ TripleComponent TurtleParser<T>::literalAndDatatypeToTripleComponentImpl(
                << ". It is treated as a plain string literal without datatype "
                   "instead."
                << std::endl;
-    parser->lastParseResult_ = std::move(literal);
+    lastParseResult_ = std::move(literal);
   };
 
   try {
     if (ad_utility::contains(integerDatatypes_, type)) {
-      parser->parseIntegerConstant(normalizedLiteralContent);
+      parseIntegerConstant(normalizedLiteralContent);
     } else if (type == XSD_BOOLEAN_TYPE) {
       if (normalizedLiteralContent == "true") {
-        parser->lastParseResult_ = true;
+        lastParseResult_ = true;
       } else if (normalizedLiteralContent == "false") {
-        parser->lastParseResult_ = false;
+        lastParseResult_ = false;
+      } else if (normalizedLiteralContent == "1") {
+        lastParseResult_ = Id::makeFromBinaryBool(true);
+      } else if (normalizedLiteralContent == "0") {
+        lastParseResult_ = Id::makeFromBinaryBool(false);
       } else {
-        makeNormalLiteral();
+        raiseOrIgnoreTriple(absl::StrCat("Invalid boolean literal: '",
+                                         normalizedLiteralContent, "'"));
```

Comment:
Can you leave this as is, 
It silently changes the observable behavior, so it at least needs to be discussed.
Please put it back for now, and bring it up in a discussion when we have more time.

Reply:
This is consistent behaviour with all other literals with datatypes. Only bool was the odd one out.